src/Stratis.Bitcoin.Features.SmartContracts.Tests/DeterminismValidationTests.cs

```diff
@@ -1037,5 +1037,157 @@ public class Test2 {
+
+            byte[] assemblyBytes = compilationResult.Compilation;
+            IContractModuleDefinition decomp = ContractDecompiler.GetModuleDefinition(assemblyBytes).Value;
+
+            var result = this.validator.Validate(decomp.ModuleDefinition);
+
+            Assert.False(result.IsValid);
+            Assert.NotEmpty(result.Errors);
+        }
+
+        [Fact]
+        public void Validate_NestedStaticField()
+        {
+            var adjustedSource = @"
+using System;
+using Stratis.SmartContracts;
+public class Test : SmartContract
+{
+    public Test(ISmartContractState state)
+        : base(state)
+    {
+    }
+
+    public struct Nested
+    {
+            static string AStaticString = ""Test"";
+    }
+}
+";
+            ContractCompilationResult compilationResult = ContractCompiler.Compile(adjustedSource);
+            Assert.True(compilationResult.Success);
+
+            byte[] assemblyBytes = compilationResult.Compilation;
+            IContractModuleDefinition decomp = ContractDecompiler.GetModuleDefinition(assemblyBytes).Value;
+
+            var result = this.validator.Validate(decomp.ModuleDefinition);
+
+            Assert.False(result.IsValid);
+            Assert.NotEmpty(result.Errors);
+        }
+
+        [Fact]
+        public void Validate_StaticMethod()
+        {
+            var adjustedSource = @"
+using System;
+using Stratis.SmartContracts;
+public class Test : SmartContract
+{
+    static string AStaticMethod() {
+        return ""Test"";
+    }
+    
+
+    public Test(ISmartContractState state)
+        : base(state)
+    {
+    }
+}
+";
+            ContractCompilationResult compilationResult = ContractCompiler.Compile(adjustedSource);
+            Assert.True(compilationResult.Success);
+
+            byte[] assemblyBytes = compilationResult.Compilation;
+            IContractModuleDefinition decomp = ContractDecompiler.GetModuleDefinition(assemblyBytes).Value;
+
+            var result = this.validator.Validate(decomp.ModuleDefinition);
+
+            Assert.True(result.IsValid);
+        }
+
+        [Fact]
+        public void Validate_StaticGenericMethod()
+        {
+            var adjustedSource = @"
+using System;
+using Stratis.SmartContracts;
+public class Test : SmartContract
+{
+    static string AStaticGenericMethod<T>() {
+        return ""Test"";
+    }
+    
+
+    public Test(ISmartContractState state)
+        : base(state)
+    {
+    }
+}
+";
+            ContractCompilationResult compilationResult = ContractCompiler.Compile(adjustedSource);
+            Assert.True(compilationResult.Success);
+
+            byte[] assemblyBytes = compilationResult.Compilation;
+            IContractModuleDefinition decomp = ContractDecompiler.GetModuleDefinition(assemblyBytes).Value;
+
+            var result = this.validator.Validate(decomp.ModuleDefinition);
+
+            Assert.False(result.IsValid);
+            Assert.NotEmpty(result.Errors);
```

Comment:
Would be nice to improve the error verification across all of these in the future, along with const strings for error messages.